### PR TITLE
Port AlphaEquivalence tests from 1.x

### PR DESCRIPTION
**Pull request by kasper-piskorski**
## What is the goal of this PR?

Port 1.x tests dealing with alpha-equivalence: `AtomicEquivalenceIT`, `AtomicQueryEquivalenceIT` for improved coverage.

## What are the changes implemented in this PR?
- ported applicable test cases from 1.x
- added symmetricity and reflexivity checking for alpha-equivalence

### CODE REVIEW

File: pattern/equivalence/AlphaEquivalenceTest.java [hunk 1]
```diff
@@ -206,48 +234,175 @@ public void test_relation_with_three_roles_equivalent() {
...
+                .collect(Collectors.toList());
+        variables.forEach(var -> testAlphaEquivalence(var.asThing(), variables, new HashSet<>()));
+    }
+
+    private void testAlphaEquivalence(ThingVariable varA, ThingVariable varB, boolean isValid){
+        assertTrue(varA.alphaEquals(varA).isValid());
+        assertTrue(varB.alphaEquals(varB).isValid());
+        assertEquals("Variable:\n" + varA + "\n=?\n" + varB, isValid, varA.alphaEquals(varB).isValid());
+        assertEquals("Variable:\n" + varB + "\n=?\n" + varA, isValid, varB.alphaEquals(varA).isValid());
+    }
```
Comment (flyingsilverfin) [inline pattern/equivalence/AlphaEquivalenceTest.java:396]: variablise so we don't do repeated work + easier to understand

File: pattern/equivalence/AlphaEquivalenceTest.java [hunk 2]
```diff
@@ -206,48 +234,175 @@ public void test_relation_with_three_roles_equivalent() {
...
+
+    private void testAlphaEquivalence(ThingVariable sourceVar, List<ThingVariable> toCheck, Set<Integer> validVars){
```
Comment (flyingsilverfin) [inline pattern/equivalence/AlphaEquivalenceTest.java:398]: can we give this a better name? is this checking alpha equivalence against itself? (reflexive?)

**Comment by flyingsilverfin** [comment]
I think @jmsfltchr  is the best person to review this - 
i forget exactly how it is right now, but we don't do "fully generic" alpha equivalence... we do an undericted alpha-equivalence walk from a starting constraint by design (it was only intended to test alpha-equivalence of `Concludable` data structures). I haven't looked too closely, i think most of the tests are fine, but I don't necessarily expect all of them to conform and pass :)

leave it here for james!

File: pattern/equivalence/AlphaEquivalenceTest.java (hunk 2 above)
Comment (flyingsilverfin) [reply pattern/equivalence/AlphaEquivalenceTest.java:398]: plus better name - `testSymmetricAlphaEquivalence` or something

**Review by jmsfltchr — CHANGES_REQUESTED** [review]
This is great, I think you tested exactly the bounds of what our alpha equivalence handles. A couple of cases to review. Also we should run these in CI, fix anything broken or `@ignore` them. We'll get you set up on Grabl @kasper-piskorski so that the tests can run there.

File: pattern/equivalence/AlphaEquivalenceTest.java [hunk 3]
```diff
@@ -206,48 +234,175 @@ public void test_relation_with_three_roles_equivalent() {
...
+                parseVariables("r1","$r1 ($x, $y) isa parentship"),
+                parseVariables("r2","$r2 ($x, child: $y) isa parentship"),
+                parseVariables("r3","$r3 (parent: $x, child: $y) isa parentship"),
+                parseVariables("r4","$r4 (father: $x, child: $y) isa parentship"),
+                parseVariables("r5","$r5 (parent: $x, daughter: $y) isa parentship"),
+                parseVariables("r6","$r6 (father: $x, daughter: $y) isa parentship")
+        )
+                .map(Variable::asThing)
+                .collect(Collectors.toList());
+        variables.forEach(var -> testAlphaEquivalenceSymmetricReflexive(var.asThing(), variables, new HashSet<>()));
```
[the remark this answers is not in the document]
Comment (kasper-piskorski) [reply pattern/equivalence/AlphaEquivalenceTest.java:280]: Yeah, this one's probably over the top.

File: pattern/equivalence/AlphaEquivalenceTest.java [hunk 4]
```diff
@@ -75,35 +87,41 @@ public void test_isa_equivalent() {
...
+    public void test_isa_different_inequivalent_variants() {
+        List<ThingVariable> variables = Stream.of(
+                parseVariables("x1", "$x1 isa organisation"),
+                parseVariables("x2", "$x2 isa company"),
+                parseVariables("y", "$y isa $type", "$type type organisation"),
+                parseVariables("z","$z isa $y")
+        )
+                .map(Variable::asThing)
+                .collect(Collectors.toList());
+        variables.forEach(var -> testAlphaEquivalenceSymmetricReflexive(var.asThing(), variables, new HashSet<>()));
```
[the remark this answers is not in the document]
Comment (kasper-piskorski) [reply pattern/equivalence/AlphaEquivalenceTest.java:104]: There is no schema backing, which is fine. Even with it how would `$x1 isa organisation` and `$x2 isa company` be alpha-equivalent? I think with tests you should assume nothing so even if it's an uninteresting case it is still useful to have.

**Comment by flyingsilverfin** [comment]
also don't forget labels and milestones and stuff :)

**Comment by jmsfltchr** [comment]
> also don't forget labels and milestones and stuff :)

Have added them so that @kasper-piskorski can see how we're doing it.

File: pattern/equivalence/AlphaEquivalenceTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (jmsfltchr) [reply pattern/equivalence/AlphaEquivalenceTest.java:104]: Ok that makes sense!

**Comment by jmsfltchr** [comment]
@kasper-piskorski can you re-run `test-behaviour`, the job failed on Grabl due to a Grabl issue. That should pass. Then, we have a couple of failures from the tests you've added. Could you look and see it the fixes are simple, implement them if so otherwise we can discuss :)

**Comment by jmsfltchr** [comment]
Note that https://github.com/graknlabs/grakn/pull/6100 may fix issues here!

**Comment by kasper-piskorski** [comment]
> Note that #6100 may fix failing tests here!

**Comment by haikalpribadi** [comment]
Why is this PR closed and not merged @kasper-piskorski ?

**Comment by kasper-piskorski** [comment]
misclicked ;p

**Review by jmsfltchr — APPROVED** [review]

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**